ethereum/core/src/main/java/org/hyperledger/besu/ethereum/chain/DefaultBlockchain.java

```diff
@@ -308,15 +323,18 @@ private BlockAddedEvent appendBlockHelper(final BlockWithReceipts blockWithRecei
     updater.putTransactionReceipts(hash, receipts);
     updater.putTotalDifficulty(hash, td);
 
-    // Update canonical chain data
-    final BlockAddedEvent blockAddedEvent = updateCanonicalChainData(updater, blockWithReceipts);
-
-    updater.commit();
-    if (blockAddedEvent.isNewCanonicalHead()) {
-      updateCacheForNewCanonicalHead(block, td);
+    final BlockAddedEvent blockAddedEvent;
+    if (storeOnly) {
+      blockAddedEvent = handleStoreOnly(blockWithReceipts);
```

Comment:
Does this mean that if/when we "rewind" to this future block in forkchoiceUpdated that we are not going to fire a an event of type `HEAD_ADVANCED` anymore?

Reply:
good catch! Indeed this need to be reviewed

Reply:
@garyschulte please check now, this commit
[Introduce forwardToBlock](https://github.com/hyperledger/besu/pull/3994/commits/b02de746dd560d68e227f108eebd8dbde214428b)

It moves head to the block sent by a previous newPayload, and triggers
the advanced head event that will also update the Bonsai world state.